### ISSUE TITLE
Computeds aren't evaluated before renders after a transaction.

### ISSUE THREAD

**Issue by dzearing** [mweststrate/mobservable#50]
Take this computed example:

```typescript
@obervable a = true;
@obervable b = false;

@observable get c() {
  return  `C thinks that the b value is: ${ b }`;
}
```

I have a react component `@observer` which renders a b and c.

I mutate them in a transaction:

```typescript
transaction(() => {
   this.a = !this.a;
   this.b = !this.b;
}
```

Expected after the transaction is complete:

c is evaluated (all computed values should be evaluated before a single render is executed)
render is called (rendering the updated c value)

Resulted:

render is called (rendering old version of c)
c is evaluated with no followup render

Explanation:

Mutating "a" causes the "render" observer to enqueue first.
Mutating "b" causes the computed observer to enqueue second.
It is a bug that the computed isn't resolved first, but even with the bug, I would have expected the render to be called twice.

If you flip the order of mutations, the issue is resolved. But this is a pretty nasty issue. The right thing to do here is to resolve all non-ui dependencies first, and then update any unique component observers that need updating second.

**Comment by mweststrate** [comment]
Hi David, 

this is the kind of evaluations that should be handled properly by Mobservable; it should construct a dependency tree that expresses that the component render uses c, so it shouldn't rerender before updating c. Can you share a bit more code? I tried to reproduce it here: http://jsfiddle.net/mweststrate/L0kd41hj/ but it seems to behave correctly.

**Comment by dzearing** [comment]
http://jsfiddle.net/L0kd41hj/4/

**Comment by kenotron** [comment]
Does mobservable actually support the property methods a la knockout?

**Comment by dzearing** [comment]
looks like it :)

I updated it here, and here's some weirdness:

The timeout does the wrong thing
The button click does the right thing

What's different?

http://jsfiddle.net/L0kd41hj/5/

**Comment by mweststrate** [comment]
@kenotron yes that is supported

@dzearing ok that's seems to be a bug indeed, I'm gonna catch some sleep and look into it! Thanks for reporting.

**Comment by mweststrate** [comment]
@dzearing The transaction inconsistency has been resolved in mobservable@1.0.7. Thanks for finding and reporting it!

**Comment by mweststrate** [comment]
Note, fixed in 1.0.8, 1.0.7 had a regression bug.
